### PR TITLE
Fix pointer scroll for nested NeverScrollables

### LINKED ISSUE #70948: Pointer scroll not handled correctly for nested NeverScrollable ScrollView
Reported in https://b.corp.google.com/issues/173650724

This appears to be a regression that happened recently, working on bisecting.

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(
    MaterialApp(
      home: Scaffold(
        appBar: AppBar(),
        body: SingleChildScrollView(
          child: Container(
            child: Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisAlignment: MainAxisAlignment.start,
              children: [
                Text(
                  'If you put your cursor over '
                  'here you can scroll    \n' * 100
                ),
                Container(
                  height: 3000,
                  width: 400,
                  child: ListView.builder(
                    physics: NeverScrollableScrollPhysics(),
                    itemCount: 100,
                    itemBuilder: (context, index) {
                      return Text('If you put your cursor here you can\'t scroll');
                    },
                  )
                ),
              ]
            )
          )
        )
      ),
    )
  );
}
```

flutter dcotor -v
```
[✓] Flutter (Channel master, 1.24.0-8.0.pre.321, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-US)
    • Flutter version 1.24.0-8.0.pre.321 at /Users/katelovett/github/flutter
    • Framework revision 05dadb0120 (14 hours ago), 2020-11-20 07:33:05 +0530
    • Engine revision a0da844845
    • Dart version 2.12.0 (build 2.12.0-50.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/katelovett/Library/Android/sdk
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C505
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (4 available)
    • iPhone 11 Pro Max (mobile) • A8FFA716-A67B-4468-B92B-F2B61F2306A0 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • Mac OS X 10.15.7 19H2 darwin-x64
    • Web Server (web)           • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 86.0.4240.198

• No issues found!
```

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This fixes a regression introduced in https://github.com/flutter/flutter/pull/66039
While initiating a pointerScroll for a Scrollable with the computed delta is correct, determining the scrollable that receives the pointer scroll event should still go through the targetScrollOffset evaluation first, otherwise in the case of the nested NeverScrollable, the event is claimed by the wrong scroll view.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/70948

## Tests

PointerScroll on nested NeverScrollable ListView goes to outer Scrollable.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by YeungKC** [comment]
When I saw the code I understood completely. It was an oversight on my part.

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 1]
```diff
@@ -630,23 +631,28 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
 
   // Returns the delta that should result from applying [event] with axis and
   // direction taken into account.
```
Comment (YeungKC) [inline packages/flutter/lib/src/widgets/scrollable.dart:633]: The position of this comment should be changed.

Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollable.dart:633]:
```suggestion
// Returns the offset that should result from applying [event] to the current
// position, taking min/max scroll extent into account.
```

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 2]
```diff
@@ -628,25 +629,30 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
...
-  // direction taken into account.
-  double _targetScrollDeltaForPointerScroll(PointerScrollEvent event) {
+  // Returns the offset that should result from applying [event] to the current
+  // position, taking min/max scroll extent into account.
+  double _targetScrollOffsetForPointerScroll(PointerScrollEvent event) {
+    final double delta = _pointerSignalEventDelta(event);
+    return math.min(math.max(position.pixels + delta, position.minScrollExtent),
+      position.maxScrollExtent);
+  }
+
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollable.dart:639]:
```suggestion

  // Returns the delta that should result from applying [event] with axis and
  // position, taking min/max scroll extent into account.	  // direction taken into account.
```

**Comment by YeungKC** [comment]
LGTM～

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 3]
```diff
@@ -628,25 +629,32 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
...
     return delta;
   }
 
   void _receivedPointerSignal(PointerSignalEvent event) {
     if (event is PointerScrollEvent && _position != null) {
-      final double targetScrollOffset = _targetScrollDeltaForPointerScroll(event);
+      final double targetScrollOffset = _targetScrollOffsetForPointerScroll(event);
       // Only express interest in the event if it would actually result in a scroll.
-      if (targetScrollOffset != 0) {
+      if (targetScrollOffset != position.pixels) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/scrollable.dart:657]: Should we be checking `_physics != null && !_physics!.shouldAcceptUserOffset(position))` before calling `pointerSignalResolver.register`?

Comment (YeungKC) [reply packages/flutter/lib/src/widgets/scrollable.dart:657]: I think that's right now because #37211

Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollable.dart:657]: That was part of another change, but I moved it to precede `pointerSignalResolver.register` and all the tests passed. Makes sense to me!

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/scrollable.dart (hunk 3 above)
Comment (YeungKC) [reply packages/flutter/lib/src/widgets/scrollable.dart:660]:
```dart
              Stack(
                children: [
                  Positioned.fill(
                    child: someList(),
                  ),
                  someList(
                      physics:NeverScrollableScrollPhysics,
                  ),
                ],
              );
```

If my pointer signal scrolls on the second list, should the expected behavior be to scroll nothing at all?

If you moved, then the first list is scrolled.

I believe this is a bit counterintuitive?

Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollable.dart:660]: This is actually a common design pattern I've seen before. In the repro provided in the bug, it creates two parallel scroll views that scroll as one. Another similar use case was reported in https://github.com/flutter/flutter/issues/70902

**Comment by hamirshekhawat** [comment]
Is this PR related to https://stackoverflow.com/questions/67706136/nested-listviews-scrolling-behavior-different-when-scrolled-using-drag-vs-scroll? I'd really like to know as the solution I have is to use Listener and MouseRegion combination to control scrollbehavior but my nested list rebuild is expensive.

**Pull request merged**